src/hydra/hydraClient.js

```diff
@@ -11,43 +11,89 @@ import isPlainObject from 'lodash.isplainobject';
 import qs from 'qs';
 import fetchHydra from './fetchHydra';
 
+class AORDocument {
+  constructor(obj) {
+    Object.assign(this, obj, {
+      originId: obj.id,
+      id: obj['@id'],
+    });
+  }
+
+  /**
+   * @return {string}
+   */
+  toString() {
+    return `[object ${this.id}]`;
+  }
+}
+
 /**
- * Transforms a JSON-LD document to an admin-on-rest compatible document
+ * Local cache containing embedded documents.
+ * It will be used to prevent useless extra HTTP query if the relation is displayed.
  *
- * @param {number} maxDepth
- * @param {number} depth
+ * @type {Map}
+ */
+const aorDocumentsCache = new Map();
+
+/**
+ * Transforms a JSON-LD document to an admin-on-rest compatible document.
+ *
+ * @param {Object} document
+ * @param {bool} clone
+ *
+ * @return {AORDocument}
  */
 export const transformJsonLdDocumentToAORDocument = (
-  maxDepth = 2,
-  depth = 1,
-) => documents => {
-  if (!isPlainObject(documents) && !Array.isArray(documents)) {
-    return documents;
+  document,
+  clone = true,
+  addToCache = true,
+) => {
+  if (clone) {
+    // deep clone documents
+    document = JSON.parse(JSON.stringify(document));
```

Comment:
meh, I hope that this is never recursive or it'll break. you could use something like [fclone](https://github.com/soyuka/fclone), not sure if needed

Reply:
It comes from a JSON, so cannot be recursive.